openlr/openlr_decoder.cpp

```diff
@@ -50,6 +69,106 @@ struct alignas(kCacheLineSize) Stats
   uint32_t m_tightOffsets = 0;
   uint32_t m_total = 0;
 };
+
+bool IsRealVertex(m2::PointD const & p, FeatureID const & fid, Index const & index)
+{
+  Index::FeaturesLoaderGuard g(index, fid.m_mwmId);
+  auto const ft = g.GetOriginalFeatureByIndex(fid.m_index);
+  bool matched = false;
+  ft->ForEachPoint(
+      [&p, &matched](m2::PointD const & fp) {
+        if (p == fp)
+          matched = true;
+      },
+      FeatureType::BEST_GEOMETRY);
+  return matched;
+};
+
+void ExpandFake(Graph::EdgeVector & path, Graph::EdgeVector::iterator edgeIt, Index const & index,
+                Graph & g)
+{
+  if (!edgeIt->IsFake())
+    return;
+
+  Graph::EdgeVector edges;
+  if (IsRealVertex(edgeIt->GetStartPoint(), edgeIt->GetFeatureId(), index))
+    g.GetRegularOutgoingEdges(edgeIt->GetStartJunction(), edges);
+  else
+    g.GetRegularIngoingEdges(edgeIt->GetEndJunction(), edges);
```

Comment:
Please, add an assertion here that edgeIt->GetEndPoint() is a real vertex?

Reply:
Done. But this is an expensive procedure to check if some vertex is a real one, it requires reading geometry.